### PR TITLE
DIV-6198: Create new case on amend for DN rejection

### PR DESCRIPTION
**Pull request by Dizrupto**
# Description

Creating a new endpoint on CMS to create a new case when petition solicitor amend a case because of DN rejection.

Fixes https://tools.hmcts.net/jira/browse/DIV-6198

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by madjava** [comment]
Just to draw your attention so Sonars messages, they seem like simple fixes.:)

File: src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/PetitionService.java [hunk 1]
```diff
@@ -28,5 +28,8 @@
 
     Map<String, Object> createAmendedPetitionDraft(String authorisation) throws DuplicateCaseException;
 
-    Map<String, Object> createAmendedPetitionDraftRefusal(String authorisation) throws DuplicateCaseException;
+    Map<String, Object> createAmendedPetitionDraftRefusalForDivorce(String authorisation) throws DuplicateCaseException;
+
+    Map<String, Object> createAmendedPetitionDraftRefusalForCCD(String authorisation, String caseId)
+        throws DuplicateCaseException;
```
Comment (Dizrupto) [inline src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/PetitionService.java:34]: Sonar is warning a 'code smell' on this as @madjava spotted.
Should we remove the throws DuplicateCaseException?
Sonar says as they are runtime exceptions, there is no benefit of adding them here.
Or do you know whether they were added for a specific purpose (e.g. informational)?
@yarpo  @jvictorcf

File: src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/impl/PetitionServiceImpl.java [hunk 2]
```diff
@@ -236,15 +256,32 @@ private CaseDetails retrieveAndValidatePetitionCase(String authorisation) throws
...
     }
 
+    private List<String> getPreviousReasonsForDivorce(Map<String, Object> caseData) {
+        List<String> previousReasons = (ArrayList<String>) caseData
+            .get(CcdCaseProperties.PREVIOUS_REASONS_DIVORCE_REFUSAL);
+
+        if (previousReasons == null) {
+            previousReasons = new ArrayList<>();
+        } else {
+            // clone to avoid updating old case
```
Comment (jacmurphy) [inline src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/impl/PetitionServiceImpl.java:278]: Could you add a logger here describing what it is doing please

File: src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/PetitionService.java (hunk 1 above)
Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/PetitionService.java:34]: Yes, I'd remove the "throws DuplicateCaseException".

Comment (jvictorcf) [reply src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/PetitionService.java:34]: To be honest, I'm not sure I agree with Sonar on this one.

Comment (Dizrupto) [reply src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/PetitionService.java:34]: Okay sure

File: src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/impl/PetitionServiceImpl.java (hunk 2 above)
Comment (Dizrupto) [reply src/main/java/uk/gov/hmcts/reform/divorce/casemaintenanceservice/service/impl/PetitionServiceImpl.java:278]: yep sure

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/petition/AmendPetitionForRefusalDraftFromCaseIdTest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.http.HttpStatus;
+import uk.gov.hmcts.reform.divorce.model.UserDetails;
+import uk.gov.hmcts.reform.divorce.support.PetitionSupport;
+
+import static org.junit.Assert.assertEquals;
+
+public class AmendPetitionForRefusalDraftFromCaseIdTest extends PetitionSupport {
+
+    @Test
+    @Ignore
```
Comment (Dizrupto) [inline src/integrationTest/java/uk/gov/hmcts/reform/divorce/petition/AmendPetitionForRefusalDraftFromCaseIdTest.java:15]: @yarpo,
This is the one causing issues along with the one below that I adjusted to what I think the author was trying to do.

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/support/CcdSubmissionSupport.java [hunk 4]
```diff
@@ -89,7 +89,7 @@ protected String getBulkCaseSubmissionRequestUrl() {
 
     Map<String, Object> getHeaders(String userToken) {
         Map<String, Object> headers = new HashMap<>();
-        headers.put(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_UTF8_VALUE);
+        headers.put(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
```
Comment (Dizrupto) [inline src/integrationTest/java/uk/gov/hmcts/reform/divorce/support/CcdSubmissionSupport.java:92]: Change made to remove deprecated warnings

File: build.gradle [hunk 5]
```diff
@@ -139,7 +139,7 @@ def versions = [
     springfoxSwagger: '2.9.2',
     springSecurityCrypto: '5.2.4.RELEASE',
     springSecurityRsa: '1.0.8.RELEASE',
-    tomcat: '9.0.35',
+    tomcat: '9.0.36',
```
Comment (yarpo) [inline build.gradle:142]: please pull the latest master

File: charts/div-cms/Chart.yaml [hunk 6]
```diff
@@ -2,4 +2,4 @@ apiVersion: v1
 appVersion: "1.0"
 description: Divorce - Case Maintenance Service
 name: div-cms
-version: 1.2.1
+version: 1.2.2
```
Comment (yarpo) [inline charts/div-cms/Chart.yaml:5]: are you sure it must be done?

File: charts/div-cms/values.yaml [hunk 7]
```diff
@@ -1,5 +1,6 @@
 java:
     applicationPort: 4010
+    aadIdentityName: divorce
```
Comment (yarpo) [inline charts/div-cms/values.yaml:3]: I think this should be done as a separate PR (I might be wrong though)

File: src/integrationTest/java/uk/gov/hmcts/reform/divorce/petition/AmendPetitionForRefusalDraftFromCaseIdTest.java (hunk 3 above)
Comment (yarpo) [reply src/integrationTest/java/uk/gov/hmcts/reform/divorce/petition/AmendPetitionForRefusalDraftFromCaseIdTest.java:15]: ok

File: build.gradle (hunk 5 above)
Comment (yarpo) [reply build.gradle:142]: oh, it's cms :)

File: charts/div-cms/Chart.yaml (hunk 6 above)
Comment (Dizrupto) [reply charts/div-cms/Chart.yaml:5]: Seems it was done automatically by a bot

File: charts/div-cms/values.yaml (hunk 7 above)
Comment (Dizrupto) [reply charts/div-cms/values.yaml:3]: The build was failing because of this. Luigi said this would fix it :)

**Pull request merged**